spec/60_show_created_by.lua

```diff
@@ -0,0 +1,29 @@
+feature("#60 Show created/updated by", function()
+    before_scenario(function()
+        when(todo, "show_button"):then_return(true)
+    end)
+
+    after_scenario(function()
+        todo.show_button:revert()
+
+        -- clear all tasks
+        global.todo.open = {}
+        global.todo.done = {}
+    end)
+
+    scenario("A task has a creator/modifier", function()
+        local player = game.players[1]
+
+        local task = todo.create_task("Test")
+        task.by = player.name
+        todo.save_task(task)
+        todo.refresh_task_table(player)
+
+        todo.create_add_edit_frame(player, task)
+        local frame = todo.get_add_edit_frame(player)
+
+        assert(frame["todo_add_task_table"]["todo_by_playername"] == player.name)
+        local text = frame["todo_add_task_table"]["todo_by_playername"]
+        faketorio.log.info("Last editor: ", {text})
+    end)
+)
```

Comment:
Missing scenario
Create task,
different player edits task
check that both attributes are shown in the edit frame.

Changing name could be done with mocking the name attribute away or just overwriting it (if possible)

Reply:
This is actually rather complicated.. I would see it as optional :)

Reply:
Create Task is done in task_spec.lua
check that both attributes are shown in the edit frame: done in "A task has a creator and a modifier" scenario
The "Different player edits task" should work.